app/src/main/java/ch/epfl/sweng/SDP/home/LeaderboardActivity.java

```diff
@@ -0,0 +1,310 @@
+import com.google.firebase.database.DataSnapshot;
+import com.google.firebase.database.DatabaseError;
+import com.google.firebase.database.ValueEventListener;
+
+import java.util.Collections;
+import java.util.LinkedList;
+
+public class LeaderboardActivity extends Activity {
+
+    private static final String TAG = "LeaderboardActivity";
+    private static final String FIREBASE_ERROR = "There was a problem with Firebase";
+    private Typeface typeMuro;
+    private LinearLayout leaderboard;
+
+    @Override
+    protected void onCreate(Bundle savedInstanceState) {
+        super.onCreate(savedInstanceState);
+        overridePendingTransition(0, 0);
+        setContentView(R.layout.activity_leaderboard);
+        leaderboard = findViewById(R.id.leaderboard);
+
+        typeMuro = Typeface.createFromAsset(getAssets(), "fonts/Muro.otf");
+
+        Glide.with(this).load(R.drawable.background_animation)
+                .into((ImageView) findViewById(R.id.backgroundAnimation));
+
+        EditText searchField = findViewById(R.id.searchField);
+
+        ((TextView) findViewById(R.id.exitButton)).setTypeface(typeMuro);
+        searchField.setTypeface(typeMuro);
+        setExitListener();
+
+        searchField.addTextChangedListener(new TextWatcher() {
+            @Override
+            public void beforeTextChanged(CharSequence query, int start, int count, int after) {
+                // Not what we need.
+            }
+
+            @Override
+            public void onTextChanged(CharSequence query, int start, int before, int count) {
+                // Not what we need.
+            }
+
+            @Override
+            public void afterTextChanged(Editable query) {
+                updateLeaderboard(query.toString());
+            }
+        });
+
+        updateLeaderboard("");
+    }
+
+    /**
+     * Sets listener and animation for exit button.
+     */
+    private void setExitListener() {
+        final TextView exit = findViewById(R.id.exitButton);
+        final Context context = this;
+        exit.setOnTouchListener(new View.OnTouchListener() {
+            @Override
+            public boolean onTouch(View view, MotionEvent event) {
+                switch (event.getAction()) {
+                    case MotionEvent.ACTION_DOWN:
+                        HomeActivity.pressButton(exit, context);
+                        break;
+                    case MotionEvent.ACTION_UP:
+                        HomeActivity.bounceButton(view, HomeActivity.MAIN_AMPLITUDE,
+                                HomeActivity.MAIN_FREQUENCY, context);
+                        launchActivity(HomeActivity.class);
+                        break;
+                    default:
+                }
+                return true;
+            }
+        });
+    }
+
+    /**
+     * Gets called when user entered a new search query.
+     * @param query new string to search
+     */
+    private void updateLeaderboard(String query) {
```

Comment:
I think that this function could be moved to a separate class **proxy** in order to increase modularity